packages/strapi-plugin-graphql/services/schema-generator.js

```diff
@@ -90,8 +90,10 @@ const generateSchema = () => {
     writeGenerateSchema(graphql.printSchema(schema));
   }
 
-  // Remove custom scalar (like Upload);
-  typeDefs = Types.removeCustomScalar(typeDefs, resolvers);
+  // Remove custom scalar (like Upload), if not using Federation
+  if (!strapi.plugins.graphql.config.federation) {
```

Comment:
Can you explain why this is only need when not using federation ? 

Also can you use _.get tu be consistent :)

Reply:
Using Federation requires the use of `buildFederatedSchema` when creating the graphql server, which means the scalar Upload needs to be defined in the schema for file uploads to still work.

I'll make the consistency change and update the comment to be a bit clearer later today.